README.md

```diff
@@ -45,8 +45,25 @@ Use Cases
 ---------
 Two common use cases include
 
-1. [Enforcing programming contracts] throughout a project via runtime checks.
-2. Producing output in `pure` procedures for debugging purposes.
+1. [Producing output in pure procedures] for debugging purposes.
+2. [Enforcing programming contracts] throughout a project via runtime checks.
+
+### Producing output in pure procedures
+Writing pure procedures communicates useful information to a compiler or a developer.
+Specifically, the pure attribute conveys compliance with several constraints that clarify data dependencies and preclude most side effects.
+For a compiler, these constraints support optimizations, including automatic parallelization on a central processing unit (CPU) or offloading to a graphics processing unit (GPU).
+For a developer, the constraints support refactoring tasks such as code movement.
+
+A developer seeking output inside a procedure presumably has an expectation regarding what ranges of output values represent correct program execution.
```

Comment:
I think we're still missing an introductory sentence for this paragraph to explain that naive print-based debugging one might expect to use is prohibited in `pure` procedures (i.e. the "problem" this subsection is purporting to solve).